inference/worker/work.py

```diff
@@ -79,6 +80,18 @@ def _prepare_message(message: inference.MessageRead) -> str:
     return prompt, parameters
 
 
+def prepare_safe_prompt(prompt: str, safety_opinion: str):
+    safety_opinion = re.sub(r"<pad>|</s>", "", safety_opinion).split("<sep>")
+    label, rots = safety_opinion[0], "and".join([x.strip(".") for x in safety_opinion[1:]])
+    if label.strip() == "__casual__":
+        return prompt
+    else:
+        pre_prompt = f"Answer the following request with {label} as responsible chatbot that believes that {rots}: "
```

Comment:
we definitely need to track when this happens and feed it back in the response to the backend, and then to the user

Reply:
also needs to be saved in the database

Reply:
(not saying we necessarily need to do this now, we can also plan to do it soon)

Reply:
Agree, especially on saving in the database. I think we can use a new form of `Response` in the inference protocol which is sent from worker to backend via the websocket in `handle_work_request`.

Reply:
I think we can log a new issue which covers:

- Communicating to the backend when the safety model intervenes
- Adding a mechanism for the backend to include safety parameters (based on frontend settings) in work requests

Reply:
yes we definitely need this. I think it's equally bad that OAI messes with your input than the fact that they don't tell you that they messed with your input